mindsdb/integrations/handlers/file_handler/tests/test_file_handler.py

```diff
@@ -0,0 +1,352 @@
+    return os.path.join(curr_dir, "data", "test.pdf")
+
+
+@pytest.fixture
+def txt_file() -> str:
+    curr_dir = os.path.dirname(os.path.realpath(__file__))
+    return os.path.join(curr_dir, "data", "test.txt")
+
+
+class TestIsItX:
+    """Tests all of the 'is_it_x()' functions to determine a file's type"""
+
+    # We can't test xlsx or parquet here because they're binary files
+    @pytest.mark.parametrize(
+        "file_path,result",
+        [(lazy_fixture("csv_file"), True), (lazy_fixture("json_file"), False)],
+    )
+    def test_is_it_csv(self, file_path, result):
+        with open(file_path, "r") as fh:
+            assert FileHandler.is_it_csv(StringIO(fh.read())) is result
+
+    @pytest.mark.parametrize(
+        "file_path,result",
+        [
+            (lazy_fixture("csv_file"), False),
+            (lazy_fixture("xlsx_file"), True),
+            (lazy_fixture("json_file"), False),
+            (lazy_fixture("parquet_file"), False),
+        ],
+    )
+    def test_is_it_xlsx(self, file_path, result):
+        assert FileHandler.is_it_xlsx(file_path) is result
+
+    # We can't test xlsx or parquet here because they're binary files
+    @pytest.mark.parametrize(
+        "file_path,result",
+        [
+            (lazy_fixture("csv_file"), False),
+            (lazy_fixture("json_file"), True),
+        ],
+    )
+    def test_is_it_json(self, file_path, result):
+        with open(file_path, "r") as fh:
+            assert FileHandler.is_it_json(StringIO(fh.read())) is result
+
+    @pytest.mark.parametrize(
+        "file_path,result",
+        [
+            (lazy_fixture("csv_file"), False),
+            (lazy_fixture("xlsx_file"), False),
+            (lazy_fixture("json_file"), False),
+            (lazy_fixture("parquet_file"), True),
+        ],
+    )
+    def test_is_it_parquet(self, file_path, result):
+        with open(file_path, "rb") as fh:
+            assert FileHandler.is_it_parquet(BytesIO(fh.read())) is result
+
+
+class TestQuery:
+    """Tests all of the scenarios relating to the query() function"""
+
+    def test_query_drop(self, monkeypatch):
+        def mock_delete(self, name):
+            return True
+
+        monkeypatch.setattr(MockFileController, "delete_file", mock_delete)
+        file_handler = FileHandler(file_controller=MockFileController())
+        response = file_handler.query(DropTables([Identifier(parts=["one"])]))
+
+        assert response.type == RESPONSE_TYPE.OK
+
+    def test_query_drop_bad_delete(self, monkeypatch):
+        def mock_delete(self, name):
+            raise Exception("File delete error")
+
+        monkeypatch.setattr(MockFileController, "delete_file", mock_delete)
+        file_handler = FileHandler(file_controller=MockFileController())
+        response = file_handler.query(DropTables([Identifier(parts=["one"])]))
+
+        assert response.type == RESPONSE_TYPE.ERROR
+
+    def test_query_select(self, csv_file):
```

Comment:
If i could get some help from @StpMax or @dusvyat or someone else:
I can't get this test to run, I always get `Exception: File 'test_data.csv' does not exists`

Reply:
looks like `test_data.csv` is missing, you can add a small csv into [here](https://github.com/mindsdb/mindsdb/tree/e4eccd25aa404539c168fb7e8c0ac1409f176a52/tests/handler_tests/data)

Reply:
@dusvyat the `csv_file` fixture is creating and saving `test_data.csv` to a temp dir, and the test sets the `content` path to be that temp dir. So the file does exist (I added some debug statements to make sure earlier), but I think maybe the 'files' integration needs to be made aware of it?